src/nodes/QskBoxRendererRect.cpp

```diff
@@ -422,28 +422,58 @@ static inline void qskCreateBorder(
     const QskBoxRenderer::Quad& out, const QskBoxRenderer::Quad& in,
     const QskBoxBorderColors& colors, Line* line )
 {
-    const Color colorLeft = colors.rgb( Qsk::Left );
-    const Color colorRight = colors.rgb( Qsk::Right );
-    const Color colorTop = colors.rgb( Qsk::Top );
-    const Color colorBottom = colors.rgb( Qsk::Bottom );
+    const QskGradient gradientLeft = colors.gradient( Qsk::Left );
+    const QskGradient gradientRight = colors.gradient( Qsk::Right );
+    const QskGradient gradientTop = colors.gradient( Qsk::Top );
+    const QskGradient gradientBottom = colors.gradient( Qsk::Bottom );
 
-    ( line++ )->setLine( in.right, in.bottom, out.right, out.bottom, colorBottom );
-    ( line++ )->setLine( in.left, in.bottom, out.left, out.bottom, colorBottom );
+    // qdebug
 
-    if ( colorLeft != colorBottom )
-        ( line++ )->setLine( in.left, in.bottom, out.left, out.bottom, colorLeft );
+    for( const QskGradientStop& stop : gradientBottom.stops() )
+    {
+        const Color c( stop.color() );
+        const qreal x1 = in.right - stop.position() * ( in.right - in.left );
```

Comment:
( in.right - in.left ) and ( out.right - out.left ) could be precalulated outside of the loop. A good compiler should do this on its own, but to be sure:

{
    const qreal y1 = in.bottom;
    const qreal y2 = out.bottom;
    const qreal dx1 = in.right - in.left;
    const qreal dx2 = out.right - out.left

    for( const QskGradientStop& stop : gradientBottom.stops() )
    {
    }
}